lib/bdev/nvme/blockdev_nvme_rpc.c

```diff
@@ -98,3 +100,100 @@ spdk_rpc_construct_nvme_bdev(struct spdk_jsonrpc_server_conn *conn,
 	free_rpc_construct_nvme(&req);
 }
 SPDK_RPC_REGISTER("construct_nvme_bdev", spdk_rpc_construct_nvme_bdev)
+
+static void
+spdk_rpc_get_nvme_devices(struct spdk_jsonrpc_server_conn *conn,
+			  const struct spdk_json_val *params,
+			  const struct spdk_json_val *id)
+{
+	struct spdk_json_write_ctx *w;
+	struct nvme_device *dev;
+	const struct spdk_nvme_ctrlr_data *cdata;
+	struct spdk_nvme_ctrlr *ctrlr;
+	struct spdk_nvme_ns *ns;
+	union spdk_nvme_vs_register vs;
+	char buf[128];
+	int i, num_ns;
+
+	if (params != NULL) {
+		spdk_jsonrpc_send_error_response(conn, id, SPDK_JSONRPC_ERROR_INVALID_PARAMS,
+						 "get_nvme_devices requires no parameters");
+		return;
+	}
+
+	if (id == NULL) {
+		return;
+	}
+
+	w = spdk_jsonrpc_begin_result(conn, id);
+	spdk_json_write_array_begin(w);
+
+	for (dev = blockdev_nvme_device_first(); dev != NULL;
+	     dev = blockdev_nvme_device_next(dev)) {
+		spdk_json_write_object_begin(w);
+
+		ctrlr = dev->ctrlr;
+		cdata = spdk_nvme_ctrlr_get_data(ctrlr);
+		vs = spdk_nvme_ctrlr_get_regs_vs(ctrlr);
+
+		snprintf(buf, sizeof(buf), "%04x:%02x:%02x.%x", dev->pci_addr.domain,
+			 dev->pci_addr.bus, dev->pci_addr.dev, dev->pci_addr.func);
+		spdk_json_write_name(w, "PCI Address");
```

Comment:
Please use lowercase with underscores names, and use the same field names as other calls where similar fields already exist (e.g. "pci_address" rather than "PCI Address").

You can also use the new spdk_json_write_string_fmt() function to avoid needing the temporary string buffer (see commit 1cb7e396a70cb0cba4d1161c1abf2e158563326b).